### PR TITLE
Acceptance wait time

### PR DESCRIPTION
**Pull request by jacksontj**
### What does this PR do?
Always enforce `acceptance_wait_time` if a retry is going to happen

### Previous Behavior
If the exception wasn't a `SaltClientError` it would spin indefinitely

### New Behavior
Wait the configured time

### CODE REVIEW

**Review by cachedout — APPROVED** [review]

File: salt/minion.py [hunk 1]
```diff
@@ -1011,10 +1011,13 @@ def _connect_minion(self, minion):
...
         Create a minion, and asynchronously connect it to a master
         '''
-        last = 0  # never have we signed in
         auth_wait = minion.opts['acceptance_wait_time']
         failed = False
         while True:
+            if failed:
+                if auth_wait < self.max_auth_wait:
+                    auth_wait += self.auth_wait
+                yield tornado.gen.sleep(auth_wait)  # TODO: log?
```
Comment (isbm) [inline salt/minion.py]: Please debug log this incident.

File: salt/minion.py [hunk 2]
```diff
@@ -3023,10 +3022,13 @@ def _connect_syndic(self, opts):
...
         Create a syndic, and asynchronously connect it to a master
         '''
-        last = 0  # never have we signed in
         auth_wait = opts['acceptance_wait_time']
         failed = False
         while True:
+            if failed:
+                if auth_wait < self.max_auth_wait:
+                    auth_wait += self.auth_wait
+                yield tornado.gen.sleep(auth_wait)  # TODO: log?
```
Comment (isbm) [inline salt/minion.py]: Yep. Debug log this please.

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by jacksontj** [comment]
I've added debug logging prior to the sleep which includes (1) what it is connecting to (2) how long it will sleep (3) what the max sleep is.

**Comment by cachedout** [comment]
@isbm Can you please review these changes and approve them if you're happy with them? Thanks.

**Review by isbm — APPROVED** [review]

**Pull request merged**